### PR TITLE
v1.10.x: Solo-kit and go-utils update

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description

Updated solo-kit and go-utils to their latest versions

# Context

go-utils bump also fixes https://github.com/solo-io/go-utils/issues/475

# Context

Resolves https://github.com/solo-io/go-utils/pull/487 in v1.10.x

### CODE REVIEW

**Comment by ben-taussig-solo** [comment]
/kick strange go mod download issue, can't identify any particular error? https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=dd104773-2c0b-46b0-89db-ea12d02a6735

File: cloudbuild.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ steps:
 
 # $COMMIT_SHA is a default gcloud env var, to run via cloudbuild submit use:
 # gcloud builds submit --substitutions COMMIT_SHA=<commit sha>,REPO_NAME=solo-io/gloo,_PR_NUM=<<insert PR Number here>> --project solo-public
-- name: 'gcr.io/$PROJECT_ID/prepare-go-workspace:0.4.15'
+- name: 'gcr.io/$PROJECT_ID/prepare-go-workspace:0.4.24'
```
Comment (sam-heilbron) [inline cloudbuild.yaml:8]: Why is this change necessary? What was the error that we would see before this change?

Comment (ben-taussig-solo) [reply cloudbuild.yaml:8]: This might not be necessary -- I was experiencing strange behavior with the CI build step, where it stopped about 1:00 and the build showed as failed. Now though, there appear to be full logs for that run. I'll try reverting the change and see if I can get CI to pass.

**Review by sam-heilbron — COMMENTED** [review]
In the original PR that bumped go-utils to pull in the logging handler npe fix (https://github.com/solo-io/gloo/pull/6709), I added logic to one of our kube2e tests to hit the logging endpoint. The idea was that this test will only pass if the request can be made. Since it's recommended in our production docs I think it's valuable that we have automated testing to confirm that it works. Can we add the same check to this branch as well?

File: cloudbuild.yaml (hunk 1 above)
Comment (ben-taussig-solo) [reply cloudbuild.yaml]: This is not necessary -- resolving this comment as I've reverted this to the original value.

**Comment by ben-taussig-solo** [comment]
> In the original PR that bumped go-utils to pull in the logging handler npe fix (#6709), I added logic to one of our kube2e tests to hit the logging endpoint. The idea was that this test will only pass if the request can be made. Since it's recommended in our production docs I think it's valuable that we have automated testing to confirm that it works. Can we add the same check to this branch as well?

Added this test in this PR: https://github.com/solo-io/gloo/pull/6807/commits/6d994c380592dd5d07d71203755cb024968cc3a5

**Review by sam-heilbron — APPROVED** [review]
Looks good!

I think in the future it would be valuable to link the PR from master that we're backporting, and outline the behavrioral change in Gloo Edge and why that is added
